request.js

```diff
@@ -92,7 +92,19 @@ function request(requestOptions, callback) {
   debug(requestId, "calledBack:", calledBack)
   var req = transport.request(options, function onResponse(res) {
     metadata = res
-    res.pipe(concat(collect))
+
+    var contentLength
+    try {
+      contentLength = parseInt(res.headers['content-length'], 10)
```

Comment:
I don't think parseInt can throw, but it can return NaN, the `=== 0` check below should probably also check for `NaN`?